Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -249,33 +249,29 @@ actor ScopeGenerator {
         let propertiesToGenerate = Set(propertiesToGenerate.compactMap(\.property))
         for propertyToGenerateNode in orderedPropertiesToGenerate {
             let hasDependenciesGeneratedByCurrentScope = !propertyToGenerateNode
-                .value
                 .requiredReceivedProperties
                 .isDisjoint(with: propertiesToGenerate)
             guard hasDependenciesGeneratedByCurrentScope else {
                 // This property does not have received dependencies generated by this scope, therefore its ordering is irrelevant.
                 continue
             }
-            var lastDependency: List<ScopeGenerator>?
-            for nextPropertyToGenerate in propertyToGenerateNode.dropFirst() {
+            for futurePropertyToGenerate in propertyToGenerateNode.dropFirst().reversed() {
```

Comment:
reversing allows us to stop iterating as soon as we hit a dependency since we know we're at the last one.